irohad/torii/impl/command_service_transport_grpc.cpp

```diff
@@ -122,13 +122,25 @@ namespace iroha {
       auto transactions = deserializeTransactions(request);
 
       auto batches = batch_parser_->parseBatches(transactions);
+      std::vector<shared_model::interface::types::HashType> unprocessed_batches;
 
       for (auto &batch : batches) {
         batch_factory_->createTransactionBatch(batch).match(
             [&](iroha::expected::Value<std::unique_ptr<
                     shared_model::interface::TransactionBatch>> &value) {
-              this->command_service_->handleTransactionBatch(
-                  std::move(value).value);
+              const shared_model::interface::types::HashType hash =
+                  value.value->reducedHash();
+              bool batch_is_processed =
+                  this->command_service_->handleTransactionBatch(
+                      std::move(value).value);
+              if (not batch_is_processed) {
+                log_->info(
+                    "Transaction batch with reduced hash {} is not processed "
+                    "and has "
+                    "to be resent",
```

Comment:
"has to be resent" is misleading, since `false` is also returned when it is a replay. Please rephrase.